src/pfe/file-watcher/server/test/functional-test/configs/project.config.ts

```diff
@@ -11,64 +11,99 @@
 /**
  * This file for different project specific (i.e liberty, spring, docker) capabaility configurations.
  */
+
+export const needManualReset: any = {
+    "docker": {
+        "runningBuildQueue": false,
+        "buildStatus": false
+    },
+    "nodejs": {
+        "runningBuildQueue": true,
+        "buildStatus": true
+    },
+};
+
 export const projectCapabilities: any = {
     "docker": {
         "startModes": ["run"],
         "controlCommands": []
+    },
+    "nodejs": {
+        "startModes": ["run", "debugNoInit"],
+        "controlCommands": ["restart"]
     }
 };
 
+export const startModes: Array<string> = ["run", "debug", "debugNoInit"];
+
 export const restartCapabilities: any = {
-    "docker": false
+    "docker": {
+        [startModes[0]]: false,
+        [startModes[1]]: false,
+        [startModes[2]]: false,
+    },
+    "nodejs": {
+        [startModes[0]]: true,
+        [startModes[1]]: false,
+        [startModes[2]]: true,
+    },
 };
 
 export const debugCapabilities: any = {
-    "docker": false
+    "docker": false,
+    "nodejs": true
 };
 
 export const mavenProfileCapabilities: any = {
     "docker":  false,
+    "nodejs": false
 };
 
-export const exposedDebugPorts: any = {
-    "docker": ["1000"]
-};
+export const exposedDebugPorts = ["1000"];
 
 export const defaultContextRoot: any = {
     "lagom": "/api/hello/you"
 };
 
 export const defaultHealthCheckEndPoint: any = {
     "go": "/",
-    "python": "/"
+    "lagom": "/",
+    "python": "/",
+    "nodejs": "/"
 };
 
 export const oneExposedPortOnly: any = {
     "go": {
         "local": true,
         "kube": true
     },
+    "lagom": {
+        "local": false,
+        "kube": true
+    },
     "python": {
         "local": true,
         "kube": true
     },
-    "lagom": {
+    "nodejs": {
         "local": false,
         "kube": true
     }
 };
 
 export const defaultInternalPorts: any = {
     "go": "8000",
+    "lagom": "9000",
     "python": "5000",
-    "lagom": "9000"
+    "nodejs": "3000"
 };
 
 export const filesToUpdate: any = {
     "go": ["Dockerfile", "main.go"],
-    "python": ["Dockerfile", "app.py"],
     "lagom": ["Dockerfile",
         "hello-api/src/main/java/com/example/rp/test/lagomendpoints/api/HelloService.java",
         "hello-impl/src/main/java/com/example/rp/test/lagomendpoints/impl/HelloModule.java",
-        "hello-impl/src/main/java/com/example/rp/test/lagomendpoints/impl/HelloServiceImpl.java"]
+        "hello-impl/src/main/java/com/example/rp/test/lagomendpoints/impl/HelloServiceImpl.java"],
+    "python": ["Dockerfile", "app.py"],
+    "nodejs": ["Dockerfile", "server/server.js"]
```

Comment:
curious, what are changing in `server/server.js`? I usually do `server/routers/health.js` for tests

Reply:
Nothing specific being changed in the file. The test basically re-writes the file and saves it and that is thought of as a file change. We can add `server/routers/health.js` in the list and that will work as well if you want.

Reply:
oh i see, so you're just relying on the socket events for build confirmation? I was thinking maybe we change file contents and verify file changes. 🤔

Reply:
The test is to verify the Turbine function works correctly and returns the correct socket event and return codes once a file change is detected.